File: nestjs-user-auth/src/utils/baseEntity.ts
export abstract class BaseEntity {
  creatorId: string;
  modifierId: string;
  deletorId: string;
  createdAt: number;
  modifiedAt: number;
  deletedAt: number;

  constructor(parial?: Partial<BaseEntity>) {
    Object.assign(this, parial);
  }

  markCreated(creatorId: string) {
    this.creatorId = creatorId;
    this.createdAt = Date.now();
    this.modifierId = creatorId;
    this.modifiedAt = Date.now();
    this.deletedAt = 0;
    this.deletorId = '';
  }

  markUpdated(modifierId: string) {
    this.modifierId = modifierId;
    this.modifiedAt = Date.now();
  }

  markDeleted(deletorId: string) {
    this.deletorId = deletorId;
    this.deletedAt = Date.now();
  }
}

File: java-order-service/src/main/java/com/example/ticket4u/pkg/entity/BaseEntity.java
package com.example.ticket4u.pkg.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Data
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
public class BaseEntity {
    private String creatorId;
    private String modifierId;
    private String deletorId;
    private int createdAt;
    private int modifiedAt;
    private int deletedAt;
}
